File: backend/src/main/java/com/sarahjewellers/model/Customer.java
package com.sarahjewellers.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "customers")
@Data
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String email;

    private String phone;

    private LocalDate birthday;

    private LocalDate anniversary;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: backend-node/models/Customer.js
const mongoose = require('mongoose');

const customerSchema = new mongoose.Schema({
  name:        { type: String, required: true },
  email:       { type: String, required: true, unique: true },
  phone:       { type: String },
  birthday:    { type: String }, // format: YYYY-MM-DD
  anniversary: { type: String }  // format: YYYY-MM-DD
}, { timestamps: true });

module.exports = mongoose.model('Customer', customerSchema);
